tenpy/algorithms/truncation.py

```diff
@@ -307,6 +307,342 @@ def svd_theta(theta, trunc_par, qtotal_LR=[None, None], inner_labels=['vR', 'vL'
+        theta_i1.itranspose(['(p1.vR)', 'vL']) # theta_i1: [(p1.vR),vL]
+        B_R, _ = npc.qr(theta_i1, inner_labels=['vL', 'vR'], inner_qconj=-1) # B_R: [(p1.vR),vL] 
+        B_R.itranspose(['vL', '(p1.vR)']) # B_R: [vL,(p1.vR)] 
+
+        theta_i0 = npc.tensordot(theta, B_R.conj(), ['(p1.vR)', '(p1*.vR*)']).ireplace_label('vL*', 'vR') # theta_i0: [(vL.p0),vR]
+        A_L, Xi = npc.qr(theta_i0, inner_labels=['vR', 'vL']) # A_L: [(vL.p0), vR]
+        
+    else:
+        # Get inital guess for the right isometry
+        Y0 = _qr_theta_Y0(old_B_L, old_B_R, theta, get_left_side, expand, min_block_increase) # Y0: [vL, (p1.vR)]
+
+        # QR based updates
+        theta_i0 = npc.tensordot(theta, Y0.conj(), ['(p1.vR)', '(p1*.vR*)']).ireplace_label('vL*', 'vR') # theta_i0: [(vL.p0),vR]
+        A_L, _ = npc.qr(theta_i0, inner_labels=['vR', 'vL']) # A_L: [(vL.p0), vR]
+
+        theta_i1 = npc.tensordot(A_L.conj(), theta, ['(vL*.p0*)', '(vL.p0)']).ireplace_label('vR*', 'vL') # theta_i1: [vL,(p1.vR)]
+        theta_i1.itranspose(['(p1.vR)', 'vL']) # theta_i1: [(p1.vR),vL]
+        B_R, Xi = npc.qr(theta_i1, inner_labels=['vL', 'vR'], inner_qconj=-1)
+        B_R.itranspose(['vL', '(p1.vR)'])
+        Xi.itranspose(['vL', 'vR'])
+
+    # SVD of bond matrix Xi
+    if use_eig_based_svd:
+        U, S, Vd, trunc_err, renormalize = _eig_based_svd(
+            Xi, need_U=get_left_side, need_Vd=(not get_left_side), inner_labels=['vR', 'vL'], trunc_params=trunc_params
+        )
+    else:
+        U, S, Vd, _, renormalize = svd_theta(Xi, trunc_params) # <- TODO: Is it fine to not specify the charges here?
+
+    # Asign return matrices
+    t_L, t_R = None, None
+    form = ['A','B']
+    if get_left_side:
+        t_L = npc.tensordot(A_L, U, ['vR', 'vL'])
+        if need_other_side:
+            if use_eig_based_svd:
+                t_R = npc.tensordot(Xi, B_R, ['vR', 'vL'])
+                t_R = npc.tensordot(U.iconj(), t_R, ['vL*', 'vL']).ireplace_label('vR*', 'vL')
+                t_R *= 1/npc.norm(t_R)
+                form[1] = 'Th'
+            else:
+                t_R = npc.tensordot(Vd, B_R, ['vR', 'vL'])
+    else:
+        t_R = npc.tensordot(Vd, B_R, ['vR', 'vL'])
+        if need_other_side:
+            if use_eig_based_svd:
+                t_L = npc.tensordot(A_L, Xi, ['vR', 'vL'])
+                t_L = npc.tensordot(t_L, Vd.iconj(), ['vR', 'vR*']).ireplace_label('vL*', 'vR')
+                t_L *= 1/npc.norm(t_L)
+                form[0] = 'Th'
+            else:
+                t_L = npc.tensordot(A_L, U, ['vR', 'vL'])
+
+    # Compute error
+    if compute_err:
+        if use_eig_based_svd:
+            theta_approx = npc.tensordot(t_L, t_R, ['vR', 'vL'])
+        else:
+            theta_approx = npc.tensordot(t_L.scale_axis(S, axis='vR'), t_R, ['vR', 'vL'])
+        eps = npc.norm(theta - theta_approx) ** 2
```

Comment:
This leads to an incorrect overlap since ``theta`` is not normalized.

```suggestion
        eps = npc.norm(theta/npc.norm(theta) - theta_approx/npc.norm(theta_approx)) ** 2
```